static/js/fenced_code.js

```diff
@@ -52,6 +52,19 @@ function wrap_quote(text) {
     return quoted_paragraphs.join('\n\n');
 }
 
+function wrap_tex(tex) {
+    try {
+        return katex.renderToString(tex, {
+            displayMode: true,
+        });
+    } catch (ex) {
+        var errspan = document.createElement('span');
+        errspan.className = 'tex-error';
+        errspan.appendChild(document.createTextNode(tex));
+        return errspan.outerHTML;
```

Comment:
You can use escape_func(tex) here to do this a bit more simply.  Can you add tests for this sort of thing:
```
~~~ math
<alert>evil</alert>
~~~
```
to make sure the tags are escaped end-to-end with both parsers?